components/question/NetPromotingScore/index.jsx

```diff
@@ -0,0 +1,66 @@
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import { useEffect, useState } from 'react';
+
+import { buildSurveyQuestionsWithAnswers } from '../Details/buildSurveyQuestionsWithAnswers';
+
+export const netPromotingScoreTestIds = {
+  base: 'rating',
+  score: 'rating__score',
+  unlikelyText: 'rating__unlikely-text',
+  likelyText: 'rating__likely-text',
+};
+
+const NetPromotingScore = ({ id, answers, setSurveyQuestionsWithAnswers }) => {
+  const [currentScore, setCurrentScore] = useState(-1);
+  const totalScore = [...Array(answers.length)];
+
+  useEffect(() => {
+    setCurrentScore(-1);
+  }, [id]);
+
+  const OnScoreChange = index => {
+    const surveyQuestionsWithAnswers = buildSurveyQuestionsWithAnswers(id, answers[index].id);
+    setCurrentScore(index);
+    setSurveyQuestionsWithAnswers(surveyQuestionWithAnswer => {
+      surveyQuestionWithAnswer = surveyQuestionWithAnswer.filter(response => response.id !== id);
+      return [...surveyQuestionWithAnswer, surveyQuestionsWithAnswers];
+    });
+  };
+
+  return (
+    <div className="w-[363px] m-auto mt-16" data-test-id={netPromotingScoreTestIds.base}>
+      <ul className="h-14 flex justify-center mb-[15.5px]">
```

Comment:
And I guess this can be `15` 🤔 

```suggestion
      <ul className="h-14 flex justify-center mb-[15px]">
```
![Screen Shot 2565-05-19 at 10 43 40](https://user-images.githubusercontent.com/1772999/169199827-0d001cb9-0ab5-4ff5-9d15-55c4ab2c4333.png)

Reply:
done in 5ae532f386553121a8c8a14be8a2ae4923c0b9b3!